### PR TITLE
Use the config provided by the back end server

### LINKED ISSUE #367: Expose available modules, extensions, and other server configurations in server endpoint
<a href="https://github.com/landonreed"></img></a> **Issue by [landonreed](https://github.com/landonreed)**
_Friday Nov 16, 2018 at 17:21 GMT_
_Originally opened as https://github.com/catalogueglobal/datatools-ui/issues/258_

----

Instead of duplicating configuration data in both the server and UI repos, use the appInfo endpoint to inform the UI of the available modules/extensions that it can utilize.

_Originally posted by @skinkie in https://github.com/catalogueglobal/datatools-ui/issues/217#issuecomment-404514565_

### PR DESCRIPTION
**Pull request by landonreed**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JSDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This all but removes the settings.yml config file in favor of requesting the config info via the
fetchAppInfo server endpoint. This action is dispatched once when any component mounts.

fix #367 refs ibi-group/datatools-server#229

### CODE REVIEW

**Comment by landonreed** [comment]
OK, @evansiroky. Comments have been addressed.

File: lib/manager/reducers/status.js [hunk 1]
```diff
@@ -195,6 +195,18 @@ const config = (state: StatusState = defaultState, action: Action): StatusState
...
+      // If config was found in the app info response, overwrite whatever was
+      // found in the UI config with the server values. This will ensure that
+      // fields/properties shared between the server and UI will stay in sync
+      // and that UI-only fields like `messages` will stay intact.
+      // Some application properties might be overwritten unnecessarily, but
+      // these should be added to the server config moving forward.
+      if (action.payload.config) {
+        window.DT_CONFIG = {
+          ...window.DT_CONFIG,
+          ...action.payload.config
```
[the remark this answers is not in the document]
Comment (evansiroky) [reply lib/manager/reducers/status.js]: The problem with overwritting the config is that it completely overwrites the gtfs spec that is defined here:

https://github.com/ibi-group/datatools-ui/blob/222e546322495b608b41fa483d9a1f28dac6a740/lib/common/util/config.js#L56-L61

When you open a feed in the editor an error occurs:

Screen Shot 2019-08-30 at 12 02 08 PM

[the remark this answers is not in the document]
Comment (landonreed) [reply lib/manager/reducers/status.js]: OK, I moved the yml files to `config.specifications` to avoid this conflict.

File: lib/types/index.js [hunk 2]
```diff
@@ -843,3 +843,52 @@ export type TripCounts = {[string]: TripCountItems} & ?{[string]: TripCounts}
...
+    }
+  },
+  messages: {
+    active: MessageFile,
+    all: Array<MessageFile>
+  },
+  modules: {
+    [name: ModuleType]: {
+      enabled?: boolean,
+      spec?: Array<any>,
```
Comment (evansiroky) [inline lib/types/index.js]: Is this needed anymore?

**Pull request merged**